webrender/src/resource_cache.rs

```diff
@@ -450,6 +460,23 @@ impl ResourceCache {
                     return;
                 }
 
+                if let Some(tile_size) = template.tiling {
+                    // If the tiling size is too big for hardware texture size then
+                    // just early out and drop the image.
+                    if tile_size as u32 > self.texture_cache.max_texture_size() {
+                        warn!("Dropping image, tile size:{} is too big for hardware!", tile_size);
+                        return;
```

Comment:
I'm a bit uncomfortable with these `return` statements as they are different from other return code paths from this method in that they are failures, while others are successes.
How about we return true/false (at least), and then also modify `prepare_prim_for_render` to return `Option<Metadata>` so that a primitive that failed to prepare can be discarded in time?

Reply:
@kvark 
I'm not familiar with the frame_builder::build process. The prepare_prim_for_render() is called by frame_builder::handle_primitive_run()[1]. And the handle_primitive_run() looks like to handle the mask and clip related thing. Are you thinking about skipping the render-task creation in handle_primitive_run() for the case of None Option<Metadata> return value? 

[1]
https://dxr.mozilla.org/mozilla-central/rev/c959327c6b75cd4930a6ea087583c38b805e7524/gfx/webrender/src/frame_builder.rs#1996

Reply:
Yes

Reply:
The prepare_prim_for_render() is changed. The "Metadata" can't be remove now.

From the first comment:

> I'm a bit uncomfortable with these return statements as they are different from other return code paths from this method in that they are failures, while others are successes.

We still have a case with a failure at [1] in request_image(). So, the new return statements are not just the new failed case. If they are still not looked good, I need to check the  prepare_prim_for_render() for the "Metadata" removing.

[1]
https://github.com/servo/webrender/blob/e11297a240b29bd57154d37ecb402d0b9aa299e6/webrender/src/resource_cache.rs#L553